### DISCUSSION TITLE
Undo content with collaboration

### DISCUSSION THREAD

**Discussion by mendesanna** [ueberdosis/tiptap#5924] [General Questions] [answered]
Hi,
I'm using Tiptap with the collaboration extension, where I save both the collaborative document and the editor content to a database.
The issue arises when saving the editor content fails. In such cases, I want to revert the user's changes to prevent the collaborative document from becoming unsynchronized with the database state.
I tried using editor.commands.undo(), but it only reverts the last change. For example, if the user makes multiple changes and the save request fails, calling undo() doesn't restore the editor to its original state—it only reverts the most recent change.
Is there a way to reset the editor content to the original state (before any changes were made) if the save operation fails?
On the other inputs with collaboration I'm currently using the undoManager which seems to work pretty well for my purpose but for tiptap I couldn't integrate that

**Comment by abecirovic3** [comment, marked as answer]
Wouldn't editor.setContent(oldJSON) work, provided that you can preserve the json.
For example if you load your document I guess initially you have the json
oldJson = json;
... updates happen (successful save)
oldJSON = json
... updates happen (failed save)
editor.setContent(oldJSON)
